V3-finalize

jaydeep-kuber opened the pull request:
Here i have made `fileWatcherV3.py`, which is more cleaner and better version of prev one. This file is present at the root of AUP.

jaydeep-brv committed:
new v3 filewatcherV3

jaydeep-brv committed:
worked on execute sql command

jaydeep-brv committed:
worked on aup v3

jaydeep-brv committed:
Stop tracking home folder

jaydeep-brv committed:
Stop tracking env folder

jaydeep-brv committed:
diffchecker and filewatcher changes in location and logging

mananbrevitaz commented on AUP-bck/AUP/playGround.py:
```diff
@@ -0,0 +1,94 @@
+        writer.writeheader()
+        writer.writerows(sortedPrevFile)
+    
+    print("########################################################################")
```
Remove these kind of prints

mananbrevitaz commented on AUP-bck/AUP/playGround.py:
```diff
@@ -0,0 +1,94 @@
+        raise ValueError(f"Column '{sort_key}' not found in CSV headers")
+    
+    sortedCurrentFile = sorted(currentFileData, key= lambda x : x[sort_key])
+    # print(sortedCurrentFile)
```
Remove this type of comments too, where we have commented the debugging print statements.

jaydeep-kuber replied:
This is a playground file. just for my practice and experiments. This is not part of Filewatcher.

mananbrevitaz commented on AUP-bck/AUP/playGround.py:
```diff
@@ -0,0 +1,94 @@
+    # creating addUpdate and disable csv files.
+
+
+sort_csv_by_column("../py/home/ubuntu/allegoAdmin/workdir/solarcity/previous.csv", "../py/home/ubuntu/allegoAdmin/workdir/solarcity/users.csv", "id")
```
Do not pass hard-coded values here. Instead pass the dynamic values

mananbrevitaz commented on AUP-bck/AUP/playGround.py:
```diff
@@ -0,0 +1,94 @@
+        writer.writerow(current_header)
+        writer.writerows(discard_rows)
+
+generate_add_discard_files("./sorted_currnt.csv","./sorted_prev.csv","./sortedFiles/add.csv","./sortedFiles/discar.csv")
```
Same as above

mananbrevitaz commented on AUP-bck/AUP/py/FwLibrary.py:
```diff
@@ -0,0 +1,176 @@
+    prod_config=Path("/home/ubuntu/allegoAdmin/scripts/prod.json")
+    
+    cid = cid
+    temp_p = Path(f"/tmp/diff_previous_{cid}.csv") 
```
Use company specific folder available in workDir

mananbrevitaz commented on AUP-bck/AUP/py/__pycache__/FwLibrary.cpython-312.pyc:
Don't include binaries in a PR it should not be pushed in the git either

mananbrevitaz commented on AUP-bck/AUP/py/home/ubuntu/allegoAdmin/workdir/solarcity/addUpdate.csv.solarcity:
Same for this kind of test files, and data files

mananbrevitaz commented on AUP/fileWatcherV3.py:
What does this V2 and V3.py files means?

mananbrevitaz commented on others/restartScholarAgent.sh:
Will ignore these for now.

jaydeep-brv committed:
static file path chages in progress

jaydeep-brv committed:
Change static paths to dynamic taking it from json/env file.

jaydeep-brv committed:
implement gpg and change in logwatcher ut

jaydeep-brv committed:
removed unwanted directories

jaydeep-brv committed:
remove py catch

jaydeep-brv committed:
gpg implementation done

mananbrevitaz commented on AUP/Crypto/gpg.py:
```diff
@@ -0,0 +1,136 @@
+
+
+def main(company: str, key: str, passphrase: str):
+    upload_dir = Path(f"/home/jay/work/scripts/AUP/home/{company}/UPLOAD")
```
Change this dir path to represent server path.

mananbrevitaz commented on AUP/fileWatcherV3.py:
```diff
@@ -0,0 +1,676 @@
+                    print(f"❌ Error during MySQL execution: {e}")
+                except Exception as e:
+                    print(f"An unexpected error occurred: {e}")
+                #     output = run_sql_script(stage_script, f"{work_directory}")
```
Remove this commented code

jaydeep-brv committed:
PR review resolve

- remove commented code from fileWatcherV3.py
- change upload and archive dir paths in gpg.py

mananbrevitaz commented on AUP/env/fwDevEnv.json:
```diff
@@ -2,9 +2,14 @@
     "number_of_company": 2,
     "log_dir": "/home/jay/work/scripts/AUP/home/ubuntu/logs",
     "allego_home": "/home/jay/work/scripts/AUP/home/ubuntu/allegoAdmin",
+    "source_parent_dir":"/home/jay/work/scripts/AUP/home",
```
update the paths with /home/ubuntu ...

mananbrevitaz commented on AUP/Crypto/encryption.py:
```diff
@@ -0,0 +1,57 @@
+    _prefix = "20250731"  # need for encryption function.
+    if len(sys.argv) > 1:
+        key = sys.argv[1]
+    upload_dir = Path(f"/home/jay/work/scripts/AUP/home/{company}/UPLOAD")
```
Update the paths here also

mananbrevitaz commented on AUP/Crypto/encryption.py:
```diff
@@ -0,0 +1,57 @@
+        return None
+
+if __name__ == "__main__":
+    company = "company_1"
```
File name should also be passed along with the key. Can you update that

mananbrevitaz commented on AUP/Crypto/gpg.py:
```diff
@@ -0,0 +1,136 @@
+        company: company name which is being processing.
+    """
+
+    print("--- THI IS PRINT LOOP --- ")
```
Typo in "THI" might be "THIS"

jaydeep-brv committed:
Reorganize: change in .gitignore file.

jaydeep-brv committed:
Reorganize: remove unwanted directory

jaydeep-brv committed:
Reorganize: remove .idea

jaydeep-brv committed:
Reorganize: remove unwanted files for UT

jaydeep-brv committed:
Reorganize: change in git ignore __pycache__

jaydeep-brv committed:
Reorganize: solve git ignore conflict

jaydeep-brv committed:
PR resolve: AUP/Crypto/gpg.py Fixed typo.
